mm2src/coins/eth.rs

```diff
@@ -2670,150 +2683,158 @@ impl EthCoin {
+        let sender =
+            try_mm_err_fus!(addr_from_raw_pubkey(sender_pub).map_to_mm(ValidatePaymentError::AddressParseError));
+        let expected_value = try_mm_err_fus!(wei_from_big_decimal(&amount, self.decimals));
         let selfi = self.clone();
         let secret_hash = secret_hash.to_vec();
         let fut = async move {
             let swap_id = selfi.etomic_swap_id(time_lock, &secret_hash);
-            let status = try_s!(
-                selfi
-                    .payment_status(expected_swap_contract_address, Token::FixedBytes(swap_id.clone()))
-                    .compat()
-                    .await
-            );
+            let status = selfi
+                .payment_status(expected_swap_contract_address, Token::FixedBytes(swap_id.clone()))
+                .compat()
+                .await
+                .map_to_mm(ValidatePaymentError::PaymentStatusError)?;
             if status != PAYMENT_STATE_SENT.into() {
-                return ERR!("Payment state is not PAYMENT_STATE_SENT, got {}", status);
+                return MmError::err(ValidatePaymentError::UnexpectedPaymentState(format!(
+                    "Payment state is not PAYMENT_STATE_SENT, got {}",
+                    status
+                )));
             }
 
-            let tx_from_rpc = try_s!(
-                selfi
-                    .web3
-                    .eth()
-                    .transaction(TransactionId::Hash(tx.hash))
-                    .compat()
-                    .await
-            );
+            let tx_from_rpc = selfi
+                .web3
+                .eth()
+                .transaction(TransactionId::Hash(tx.hash))
+                .compat()
+                .await?;
             let tx_from_rpc = match tx_from_rpc {
                 Some(t) => t,
-                None => return ERR!("Didn't find provided tx {:?} on ETH node", tx),
+                None => {
+                    return MmError::err(ValidatePaymentError::MissingTx(format!(
+                        "Didn't find provided tx {:?} on ETH node",
+                        tx
+                    )))
+                },
             };
 
             if tx_from_rpc.from != sender {
-                return ERR!(
+                return MmError::err(ValidatePaymentError::WrongSenderAddress(format!(
                     "Payment tx {:?} was sent from wrong address, expected {:?}",
-                    tx_from_rpc,
-                    sender
-                );
+                    tx_from_rpc, sender
+                )));
             }
 
             match &selfi.coin_type {
                 EthCoinType::Eth => {
                     if tx_from_rpc.to != Some(expected_swap_contract_address) {
-                        return ERR!(
+                        return MmError::err(ValidatePaymentError::WrongReceiverAddress(format!(
                             "Payment tx {:?} was sent to wrong address, expected {:?}",
```

Comment:
I think we can change the `ValidatePaymentError::WrongReceiverAddress` signature:
```rust
pub enum ValidatePaymentError {
  #[display(fmt = "Payment tx was sent to wrong address {:?}, expected {:?}", found, expected)]
  WrongReceiverAddress {
    found: String,
    expected: String,
  }
}
```
The same for `WrongSenderAddress`